### PR TITLE
raise err in io.NDArrayIter for invalid usecase

### PR DESCRIPTION
**Pull request by ZiyueHuang**
## Description ##
fix https://github.com/apache/incubator-mxnet/issues/9211

cc @eric-haibin-lin @dingran

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] unittest

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/io.py [hunk 1]
```diff
@@ -515,17 +515,10 @@ def _init_data(data, allow_empty, default_name):
...
-        for v in data:
-            if isinstance(v, dtype):
-                return True
-    if isinstance(data, dict):
-        for v in data.values():
-            if isinstance(v, dtype):
-                return True
+    """return True if data has instance of dtype after _init_data"""
+    for item in data:
+        if isinstance(item[1], dtype):
```
Comment (eric-haibin-lin) [inline python/mxnet/io.py]: why `item[1] `?

Comment (ZiyueHuang) [reply python/mxnet/io.py]: This function is called after _init_data which transforms the data to [ ( name, ndarray), ... ] format.

Comment (eric-haibin-lin) [reply python/mxnet/io.py]: I see. Do you mind updating the comment to note the format of the input? It's not obvious unless someone read the code for _init_data

Comment (szha) [reply python/mxnet/io.py]: One option to improve readability:
```
name, arr = item[1]
if isinstance(...):
```

**Pull request merged**